### ISSUE TITLE
Return `found` and `out_of` for facet counts so UI can show total number of facets

### ISSUE THREAD

**Issue by nojvek** [typesense/typesense#273]
## Description

By passing `max_facet_values`, there is no way to know in the UI how many total values there are. We only know that the server returned max 20. Without this it's not possible to build an experience where the UI says "Showing top X of Y" in facet list.

You have to resort to a generic "Show more/show less"

![image](https://user-images.githubusercontent.com/1018196/118916738-6a46a100-b8e4-11eb-91c1-341a8b8b7828.png)

## Steps to reproduce

1. Go to https://songs-search.typesense.org/?s%5Bquery%5D=s&s%5BrefinementList%5D%5Bprimary_artist_name%5D%5B0%5D=Ross%20Hardy
2. Look at 
```
curl 'https://qtg5aekc2iosjh93p.a1.typesense.net/multi_search?x-typesense-api-key=8hLCPSQTYcBuK29zY5q6Xhin7ONxHy99' \
  -X POST \
  --data-raw '{"searches":[{"query_by":"title,primary_artist_name,album_name","query_by_weights":"2,2,1","sort_by":"","highlight_full_fields":"title,primary_artist_name,album_name","collection":"s","q":"s","facet_by":"genres,primary_artist_name,release_group_types,country,release_decade","filter_by":"primary_artist_name:=[Ross Hardy]","max_facet_values":20,"page":1,"per_page":15},{"query_by":"title,primary_artist_name,album_name","query_by_weights":"2,2,1","sort_by":"","highlight_full_fields":"title,primary_artist_name,album_name","collection":"s","q":"s","facet_by":"primary_artist_name","filter_by":"","max_facet_values":20,"page":1,"per_page":1}]}' 
 ```

## Expected Behavior

The returned object comes back as 
```
facet_counts: [
  {field_name: "country", counts: [...], stats: {}, found: 20, out_of: 50}, 
  ...
]
```

## Actual Behavior

![image](https://user-images.githubusercontent.com/1018196/118916414-be9d5100-b8e3-11eb-8c40-6f886dc5c437.png)

## Additional Context

Also would be nice to specify `max_facet_values` per field, rather than one global number.

**Comment by kishorenc** [comment]
I can certainly see how it would be useful but it is tricky because facet computation is a very expensive operation, which is parallelized as independent sub-tasks operating on parts of the data. To fetch the total facet values we have to bring all the unique values from each sub-task into a single task and calculate a unique count. This will be very slow on large data sets.

**Comment by nojvek** [comment]
Could it be at-least an option? i.e for small datasets we can choose to get accurate count, but for large datasets we use an approx count?

**Comment by kishorenc** [comment]
Open to making this optional, adding it to our backlog, targeting hopefully next release. I will keep this thread posted.

**Comment by kishorenc** [comment]
@nojvek @sanchaz 

I've got a RC Docker build where this is available: `typesense/typesense:0.22.0.rcs4` 

The total count is now returned as `total_facet_values` in the `stats` field:

<img width="327" alt="Screenshot 2021-09-10 at 7 18 32 PM" src="https://user-images.githubusercontent.com/782073/132863613-4af73327-c0fc-4c2b-b253-1fa95c15723c.png">

**Comment by jasonbosco** [comment]
Released in v0.22
